server/src/main/java/org/candlepin/async/tasks/HealEntireOrgJob.java

```diff
@@ -72,18 +71,23 @@ public HealEntireOrgJob(Entitler entitler, ConsumerCurator consumerCurator, Owne
     public Object execute(JobExecutionContext context) throws JobExecutionException {
         try {
 
-            final JobArguments arguments = context.getJobArguments();
-            final String ownerKey = arguments.getAsString(OWNER_KEY);
+            JobArguments arguments = context.getJobArguments();
+            String ownerKey = arguments.getAsString(OWNER_KEY);
             Owner owner = ownerCurator.getByKey(ownerKey);
+            if (owner == null) {
+                log.warn("Healing attempted against non-existent org id \"{}\"", ownerKey);
+                throw new JobExecutionException(
+                    i18n.tr("Healing attempted against non-existent org id \"{}\"", ownerKey));
```

Comment:
Minor nit: the key and ID are different values, so this error message might be a bit confusing.

Also, need the terminal flag here as well.

Reply:
Done.